components/retro-go/rg_audio.c

```diff
@@ -236,6 +255,8 @@ void rg_audio_submit(const rg_audio_frame_t *frames, size_t count)
 
     if (audio.sink->type == RG_AUDIO_SINK_DUMMY)
     {
+        // BUG: this usleeps too long because it doesn't take into account the time that was spent drawing the frame
+        // before submitting the audio, thus making the emulator run too slow, although you can speed it up in the settings.
         rg_usleep((uint32_t)(count * (1000000.f / audio.sampleRate)));
```

Comment:
This is a deliberate choice! It used to work as you describe and take into account frame time (check the file history), but for profiling purposes this was better.

I tend to agree that we should have a dummy driver that is time-accurate for devices that simply have no other audio drivers, though. Maybe have a separate sink for debugging purposes.

You can leave the comment as a reminder that I should sort this out :)